### PR TITLE
GitHub: resolve workflow & webhook IDs for approval summaries (issue #973 phase 3)

### PR DESCRIPTION
**Pull request by chiedocursor**
Implements **Phase 3** of #973: add `ResourceDetailResolver` for the GitHub connector so approval summaries show human-readable workflow names and webhook URLs instead of raw IDs.

## Summary

- New `connectors/github/resolve_resource_details.go`: `github.trigger_workflow` → GET `/repos/{owner}/{repo}/actions/workflows/{workflow_id}` → `workflow_name`; `github.delete_webhook` → GET `/repos/{owner}/{repo}/hooks/{hook_id}` → `webhook_url`, `webhook_events` (comma-separated).
- Updated display templates in `manifest.go` to use `{{workflow_name}}` and `{{webhook_url}}`.
- Registered `workflow_name`, `webhook_url`, and `webhook_events` in `connectors/display_template_test.go`.
- Added `resolve_resource_details_test.go` with httptest coverage.

`GitHubConnector` now satisfies `connectors.ResourceDetailResolver` (same pattern as Slack/Google); agent approvals already pick this up via type assertion in `api/agent_approvals.go`.

## Phase 3 checklist (issue #973)

- [x] Create `connectors/github/resolve_resource_details.go` implementing the `ResourceDetailResolver` interface (follow Slack/Google pattern)
- [x] Implement `resolveWorkflow()` for `github.trigger_workflow` — `GET /repos/{owner}/{repo}/actions/workflows/{workflow_id}` → `{"workflow_name": "..."}`
- [x] Implement `resolveWebhook()` for `github.delete_webhook` — `GET /repos/{owner}/{repo}/hooks/{hook_id}` → `{"webhook_url": "...", "webhook_events": "push, pull_request"}`
- [x] Update `github.trigger_workflow` `DisplayTemplate` (manifest.go:596) to use `{{workflow_name}}`
- [x] Update `github.delete_webhook` `DisplayTemplate` (manifest.go:1270) to use `{{webhook_url}}`
- [x] Add `workflow_name`, `webhook_url`, `webhook_events` to known fields in `display_template_test.go`
- [x] Create `connectors/github/resolve_resource_details_test.go` with coverage for both resolvers
- [x] Wire the resolver into the connector's registration so `ResolveResourceDetails` is called (check how Slack/Google expose it via the connector struct)

Refs #973 (partial — phases 1, 2, and 4 are separate PRs).

## Test plan

- [ ] `make test-backend` (not run in this environment: local Go toolchain is older than `go.mod`; CI should run the full suite)
- [x] `gofmt -e` on new/edited Go files
- [ ] `make test-frontend` / `make mobile-test` (no frontend/mobile changes)

<div><a href="https://cursor.com/agents/bc-4b73b006-55cb-4010-8785-52c52ba4d34a"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-4b73b006-55cb-4010-8785-52c52ba4d34a"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by chiedocursor** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
No review comments acted on.

---
*Watch session ended after 0 minutes of inactivity. Processed 0 comments across 1 poll cycles.*

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 1/6

**Findings:** 5 (0 critical, 1 high, 1 medium, 3 low)

Clean implementation that closely mirrors the established Slack/Google pattern — compile-time interface assertion, proper URL encoding, reuse of existing validation helpers, and the display-template test registry is updated alongside the new fields. Test setup with `newForTest` + `httptest.Server` is tidy.

The main concern is a UX regression in the new display templates: the old templates always showed an identifier (`{{workflow_id}}`, `{{hook_id}}`) from the parameters themselves, but the new templates rely entirely on resolver output. When resolution fails (stale token, insufficient scopes, deleted resource, 5s timeout, etc.), the approval is still stored — now with `resource_details = null` — and the template renders as `"Delete webhook  in owner/repo"` / `"Trigger workflow  in owner/repo"` with no identifier. For `delete_webhook` this is particularly problematic since it's a `high` risk action and the approver is left with literally no information about which webhook is being deleted.

A few lower-priority items around error-type consistency and test coverage of the graceful-degradation branches (empty name / empty hook config).

### What Looks Good
- `var _ connectors.ResourceDetailResolver = (*GitHubConnector)(nil)` — great compile-time guard, same as Slack/Google.
- Reuses `requireOwnerRepo` / `requirePositiveInt` instead of re-deriving validation inline.
- Correct use of `url.PathEscape` for all three path segments (owner, repo, workflow_id).
- `strings.TrimSpace` guard against whitespace-only names is a nice touch, and returning `nil, nil` when there's nothing worth showing is the right shape for the caller.
- Test registry update in `display_template_test.go` keeps `TestDisplayTemplateParamsExist` passing and leaves a clear breadcrumb comment for future connectors.
- All tests use `t.Parallel()`.

---
*Automated review by `/review` — Round 1 of 6*

---
_Generated by [Claude Code](https://claude.ai/code)_

File: connectors/github/manifest.go [hunk 1]
```diff
@@ -1267,7 +1267,7 @@ func (c *GitHubConnector) Manifest() *connectors.ConnectorManifest {
 				Name:            "Delete Webhook",
 				Description:     "Delete a repository webhook by numeric hook ID",
 				RiskLevel:       "high",
-				DisplayTemplate: "Delete webhook {{hook_id}} in {{owner}}/{{repo}}",
+				DisplayTemplate: "Delete webhook {{webhook_url}} in {{owner}}/{{repo}}",
```
Comment (chiedoclaude) [inline connectors/github/manifest.go:1270]: **[High · UX/Architecture]** This template has no fallback identifier — if `ResolveResourceDetails` fails (webhook already deleted, token missing `admin:repo_hook`, 5s timeout, 5xx from GitHub), `resource_details` is stored as null, and the template renders as `"Delete webhook  in acme/app"`. The approver is left with zero information about which webhook is about to be deleted — for a `high`-risk destructive action.

**Why it matters:** Approvers may rubber-stamp a destructive action without the context needed to catch a wrong target. The old template (`{{hook_id}}`) always had a stable identifier from the params themselves.

**Fix — two options:**

1. Include both in the template so the ID is always visible:
   ```go
   DisplayTemplate: "Delete webhook {{webhook_url}} (#{{hook_id}}) in {{owner}}/{{repo}}",
   ```
   This requires adding `hook_id` to `resourceDetailFields` in `display_template_test.go` *or* keeping it in the params schema (it already is — so no test change needed; the schema already lists `hook_id`).

2. Have `resolveWebhook` populate `webhook_url` with a fallback string like `fmt.Sprintf("#%d", p.HookID)` when the URL lookup fails. Keeps the template short but hides the raw ID when resolution succeeds.

Option 1 is more explicit and matches how you'd want a high-risk action surfaced.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: connectors/github/manifest.go [hunk 2]
```diff
@@ -593,7 +593,7 @@ func (c *GitHubConnector) Manifest() *connectors.ConnectorManifest {
 				Name:            "Trigger Workflow",
 				Description:     "Trigger a GitHub Actions workflow dispatch event",
 				RiskLevel:       "medium",
-				DisplayTemplate: "Trigger workflow {{workflow_id}} in {{owner}}/{{repo}}",
+				DisplayTemplate: "Trigger workflow {{workflow_name}} in {{owner}}/{{repo}}",
```
Comment (chiedoclaude) [inline connectors/github/manifest.go:596]: **[Medium · UX]** Same concern as the `delete_webhook` template below: if workflow-name resolution fails (repo is private and the resolver credentials lack `actions:read`, workflow was renamed/deleted, network error, 5s timeout), this renders as `"Trigger workflow  in acme/app"` — worse than the previous `{{workflow_id}}` template, which at least showed `deploy.yml` or the numeric ID.

**Why it matters:** Medium risk so the blast radius is smaller, but approvers still lose the only identifier when the happy path breaks.

**Fix:**
```go
DisplayTemplate: "Trigger workflow {{workflow_name}} ({{workflow_id}}) in {{owner}}/{{repo}}",
```
`workflow_id` is already in the schema, so no `resourceDetailFields` change needed.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: connectors/github/resolve_resource_details.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		WorkflowID string `json:"workflow_id"`
+	}
+	if err := json.Unmarshal(params, &p); err != nil {
+		return nil, fmt.Errorf("parse params: %w", err)
+	}
+	if err := requireOwnerRepo(p.Owner, p.Repo); err != nil {
+		return nil, err
+	}
+	if p.WorkflowID == "" {
+		return nil, fmt.Errorf("missing workflow_id")
```
Comment (chiedoclaude) [inline connectors/github/resolve_resource_details.go:43]: **[Low · Code Quality]** Inconsistent error types: `requireOwnerRepo` above returns a `*connectors.ValidationError`, but this manual check returns a plain `fmt.Errorf`. The same inconsistency is on L74 for `hook_id` (though there the existing helper `requirePositiveInt` is used — good).

Compare with the action validation in `trigger_workflow.go:32`:
```go
return &connectors.ValidationError{Message: "missing required parameter: workflow_id"}
```

**Why it matters:** Callers that type-switch on error (e.g., to distinguish user-input errors from external/network errors in telemetry) will misclassify this one. Minor — since the resolver always runs *after* schema validation in `agent_approvals.go`, this branch is effectively unreachable in production. But the test exercises it, so worth making it consistent.

**Fix:**
```go
if p.WorkflowID == "" {
    return nil, &connectors.ValidationError{Message: "missing required parameter: workflow_id"}
}
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: connectors/github/resolve_resource_details.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	if err := c.do(ctx, creds, http.MethodGet, path, nil, &resp); err != nil {
+		return nil, err
+	}
+
+	out := map[string]any{}
+	if u := strings.TrimSpace(resp.Config.URL); u != "" {
+		out["webhook_url"] = u
+	}
+	if len(resp.Events) > 0 {
+		out["webhook_events"] = strings.Join(resp.Events, ", ")
```
Comment (chiedoclaude) [inline connectors/github/resolve_resource_details.go:95]: **[Low · Performance/UX]** Webhooks can be configured with many events (GitHub supports 50+ event types, and `events: ["*"]` is common). Joining all of them with `", "` can produce template values longer than the 60-char truncation applied by `formatHighlightValue` on the frontend — which will chop the list mid-event-name (e.g., `"push, pull_request, release, issue_comm…"`).

**Why it matters:** Minor readability issue; the UI handles truncation but the result looks awkward. Nothing functionally broken.

**Fix (optional):** cap to the first few events with an overflow suffix, similar to `formatHighlightValue`'s own array formatting:
```go
if n := len(resp.Events); n > 0 {
    if n <= 3 {
        out["webhook_events"] = strings.Join(resp.Events, ", ")
    } else {
        out["webhook_events"] = fmt.Sprintf("%s, +%d more", strings.Join(resp.Events[:3], ", "), n-3)
    }
}
```
Or leave as-is and let the frontend truncate — equally defensible.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: connectors/github/resolve_resource_details_test.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+func TestResolveResourceDetails_WorkflowMissingID(t *testing.T) {
+	t.Parallel()
+
+	conn := New()
+	params, _ := json.Marshal(map[string]string{"owner": "a", "repo": "b", "ref": "main"})
+	_, err := conn.ResolveResourceDetails(context.Background(), "github.trigger_workflow", params, validCreds())
+	if err == nil {
+		t.Fatal("expected error for missing workflow_id")
+	}
+}
```
Comment (chiedoclaude) [inline connectors/github/resolve_resource_details_test.go:108]: **[Low · Test Coverage]** Two graceful-degradation branches in the resolver are uncovered:

- `resolve_resource_details.go:55-57` — `resp.Name` empty/whitespace → returns `nil, nil` (no error).
- `resolve_resource_details.go:97-99` — webhook response with neither `config.url` nor `events` → returns `nil, nil`.

**Why it matters:** These branches are the "success but nothing to show" path. Without tests, a future refactor could accidentally turn them into errors (which would get logged + Sentry'd every time a webhook happens to have empty config). For comparison, Google's test suite covers similar empty-response paths.

**Fix:** add something like
```go
func TestResolveResourceDetails_WebhookEmptyResponse(t *testing.T) {
    t.Parallel()
    srv, conn := testGitHubResolveServer(t, func(w http.ResponseWriter, _ *http.Request) {
        w.Header().Set("Content-Type", "application/json")
        _, _ = w.Write([]byte(`{}`))
    })
    defer srv.Close()

    params, _ := json.Marshal(map[string]any{"owner": "a", "repo": "b", "hook_id": 1})
    details, err := conn.ResolveResourceDetails(context.Background(), "github.delete_webhook", params, validCreds())
    if err != nil {
        t.Fatalf("unexpected error: %v", err)
    }
    if details != nil {
        t.Errorf("expected nil details for empty response, got %v", details)
    }
}
```
An equivalent test for `resolveWorkflow` with `{"name":""}` covers the other branch. Also consider a 404/500 test (Google has one at `resolve_resource_details_test.go:223`) to confirm API errors surface and don't panic.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by chiedoclaude — COMMENTED** [review]
Inline findings for Round 1 — see the earlier summary review for the overall assessment and the "What Looks Good" notes. 5 inline comments below (1 High, 1 Medium, 3 Low).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by chiedoclaude** [comment]
## Code Review Complete

**PR:** #982 — GitHub: resolve workflow & webhook IDs for approval summaries (issue #973 phase 3)
**Author:** @chiedocursor
**Rounds completed:** 1/6 (+1 skipped clean round)
**Reason for stopping:** PR merged

### Outstanding Issues

The PR merged before the findings were addressed. Carrying forward for a follow-up:

- **High** — `delete_webhook` display template (`connectors/github/manifest.go:1270`) has no identifier fallback. When `ResolveResourceDetails` fails the approval renders as `"Delete webhook  in owner/repo"` — zero context for a `high`-risk destructive action. Fix: include `{{hook_id}}` in the template, e.g. `"Delete webhook {{webhook_url}} (#{{hook_id}}) in {{owner}}/{{repo}}"`.
- **Medium** — `trigger_workflow` template (`connectors/github/manifest.go:596`) has the same gap. Fix: `"Trigger workflow {{workflow_name}} ({{workflow_id}}) in {{owner}}/{{repo}}"`.
- **Low** — `resolve_resource_details.go:43`: `fmt.Errorf("missing workflow_id")` should be `&connectors.ValidationError{...}` for consistency with the rest of the connector and with `requireOwnerRepo`.
- **Low** — Unbounded event-list join in `resolveWebhook` can produce awkwardly truncated strings when a webhook has many events. Consider capping at ~3 with `+N more`.
- **Low** — Missing test coverage for the `nil, nil` graceful-degradation branches (`resolve_resource_details.go:55-57` and `:97-99`) and for API-error paths (404/500). Google has equivalent tests at `connectors/google/resolve_resource_details_test.go:223`.

These are straightforward to address in a follow-up PR — happy to open one if helpful.

### Review History
| Round | Inline Comments | Notes |
|-------|----------------|-------|
| 1 | 5 | Initial review: 1 High, 1 Medium, 3 Low. |
| 2 | 0 | No new commits — skipped. |
| 3 | — | PR merged during wait before this round. |

### Overall Assessment

Clean, idiomatic implementation that mirrors the existing Slack/Google `ResourceDetailResolver` pattern closely — compile-time interface assertion, correct URL encoding, reuse of shared validation helpers, and the template-field test registry was updated alongside the new keys. Tests are well-structured with `t.Parallel()` and `httptest.Server`.

The one substantive concern is the display-template regression: the previous templates always showed an identifier from the action parameters, but the new ones depend entirely on successful resolution. For a `high`-risk action like `delete_webhook`, a failed resolution produces a rendered summary with no identifier at all — a real UX/safety gap for approvers. Worth a small follow-up to restore the ID fallback.

---
*Review performed by Claude Code `/review` skill*

---
_Generated by [Claude Code](https://claude.ai/code)_
